main/resource-access.bs

```diff
@@ -5,28 +5,32 @@ Issue: Write introduction to the Authenticated Resource Access section.
 
 ## Hypertext Transfer Protocol ## {#http}
 
-Issue: Write Hypertext Transfer Protocol section.
-
-Draft:
-A Solid data pod MUST implement HTTP/1.1 Message Syntax and Routing [[!RFC7230]]
-
-Draft:
-A Solid data pod MUST implement HTTP/1.1 Semantics and Content [[!RFC7231]].
-
-Draft:
-A Solid data pod MUST implement HTTP/1.1 Authentication [[!RFC7235]].
-
-Draft:
-A Solid data pod SHOULD implement HTTP/1.1 Conditional Requests [[!RFC7232]].
-
-Draft:
-A Solid data pod SHOULD implement HTTP/1.1 Range Requests [[!RFC7233]].
-
-Draft:
-A Solid data pod SHOULD implement HTTP/1.1 Caching [[!RFC7234]].
-
-Draft:
-A Solid data pod MAY implement HTTP/2 [[!RFC7540]].
+A [=data pod=] MUST be an HTTP/1.1 server [[!RFC7230]][[!RFC7231]].
+It SHOULD additionally implement HTTP/2 [[!RFC7540]]
+to improve performance,
+especially in cases where individual clients
+are expected to send high numbers of successive requests.
+
+A data pod SHOULD use the `https` URI scheme
+in order to secure the connection between clients and servers.
+When both `http` and `https` are supported,
```

Comment:
I wonder if a SHOULD here is not strong enough? How do we feel about requiring `https` at least for key API endpoints?

Reply:
See here: https://github.com/solid/specification/pull/26/files#diff-8d72b2ee1631b8eec737030e9b4a3c9c

MUST would also mean development/testing mode, and behind a reverse proxy.

Reply:
(I think you meant to include a line number in that link?)

I agree, making things easier for dev and testing is an important goal. But for what it's worth, the current implementations (node-solid-server) _require_ TLS (at least behind a self-signed cert). As in, they won't run any sort of WebID-related API endpoints without it.

Reply:
(It should point to all changes in main/security.bs)

> But for what it's worth, the current implementations (node-solid-server) _require_ TLS (at least behind a self-signed cert).

They do not actually; I am running NSS in HTTP mode behind a reverse proxy that terminates HTTPS.